### PR TITLE
Corrected bug on empty subject of com_mailto

### PR DESCRIPTION
**Pull request by emmanuel-tsalampounis**
### Summary of Changes
Corrected the problem that if the subject is empty, the default subject text is never added.

### Testing Instructions
A com_mailto request without subject.

### Expected result
The default text to be taken.

### Actual result
Empty subject.

### Documentation Changes Required
Some changes are needed at: components\com_mailto\controller.php

### CODE REVIEW

File: components/com_mailto/controller.php [hunk 1]
```diff
@@ -110,7 +110,7 @@ public function send()
 		$sender          = $this->input->post->getString('sender', '');
 		$from            = $this->input->post->getString('from', '');
 		$subject_default = JText::sprintf('COM_MAILTO_SENT_BY', $sender);
-		$subject         = $this->input->post->getString('subject', $subject_default);
+		$subject         = $this->input->post->getString('subject') !== "" ? $this->input->post->getString('subject') : $subject_default;
```
Comment (ggppdk) [inline components/com_mailto/controller.php]: I see what you are trying to fix

just you are now breaking a case that worked before, thus introducing a bug

-- that is the case that subject is NULL, aka not-set (not submitted / posted)

so better use the following code to catch both cases of empty string but also case of not set / not posted (NULL) ( the below code will disallow case of subject '0' but that is ok )) ):
```php
$subject         = $this->input->post->getString('subject') ?: $subject_default;
```

Comment (emmanuel-tsalampounis) [reply components/com_mailto/controller.php]: Please check my new commit that also catches null result.

**Review by ggppdk — APPROVED** [review]

**Comment by ggppdk** [comment]
I have tested this item :white_check_mark: successfully on 6c9f5380005d0f2f3682039bf31141bf2c6e5808<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19956">issues.joomla.org/tracker/joomla-cms/19956</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 6c9f5380005d0f2f3682039bf31141bf2c6e5808<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19956">issues.joomla.org/tracker/joomla-cms/19956</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19956">issues.joomla.org/tracker/joomla-cms/19956</a>.</sub>

**Pull request merged**